applications/Chat/coati/trainer/base.py

```diff
@@ -186,3 +187,20 @@ def fit(
                         self._update_phase(update_step)
                     # NOTE: this is for on-policy algorithms
                     self.data_buffer.clear()
+                if is_rank_0() and (episode + 1) % save_per_num_episodes == 0:
+                    if args.lora_rank > 0 and args.merge_lora_weights:
+                        from coati.models.lora import LORA_MANAGER
+
+                        # NOTE: set model to eval to merge LoRA weights
+                        LORA_MANAGER.merge_weights = True
```

Comment:
I think `merge_weights=True` should be set as a default behavior. The merged weights are much larger than lora weights. Most users only want to save lora weights. We can provide a script to merge them offline.